File: type-script-nest-graph-ql-실습/src/app/people/model/dto/response/people.response.dto.ts
import { Sex } from "../../../../../common/enum/people.sex.enum";
import { BloodType } from "../../../../../common/enum/people.blood-type.enum";
import { Role } from "../../../../../common/enum/people.role.enum";
import { PeopleEntity } from "../../entities/people.entity";

export class PeopleResponseDto {
  private peopleId: number;
  private teamId: number;
  private lastName: string;
  private firstName: string;
  private sex: Sex;
  private bloodType: BloodType;
  private serveYears: number;
  private role: Role;
  private hometown: string;

  constructor(peopleEntity: PeopleEntity) {
    this.peopleId = peopleEntity.peopleId;
    this.teamId = peopleEntity.team.teamId;
    this.lastName = peopleEntity.lastName;
    this.firstName = peopleEntity.firstName;
    this.sex = peopleEntity.sex;
    this.bloodType = peopleEntity.bloodType;
    this.serveYears = peopleEntity.serveYears;
    this.role = peopleEntity.role;
    this.hometown = peopleEntity.hometown;
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/people/model/dto/response/PeopleResponseDto.kt
package com.junyharangstudy.kotlingraphqltest.api.people.model.dto.response

import com.junyharangstudy.kotlingraphqltest.api.common.enumtype.BloodType
import com.junyharangstudy.kotlingraphqltest.api.common.enumtype.Role
import com.junyharangstudy.kotlingraphqltest.api.common.enumtype.Sex
import com.junyharangstudy.kotlingraphqltest.api.people.model.entity.People

class PeopleResponseDto (
    private val peopleId: Long,
    private val teamId: Long,
    private val lastName: String,
    private val firstName: String,
    private val sex: Sex,
    private val bloodType: BloodType,
    private val serveYears: Int,
    private val role: Role,
    private val hometown: String
) {
    companion object {
        fun peopleToDto(people: People): PeopleResponseDto {
            return PeopleResponseDto(
                peopleId = people.peopleId!!,
                teamId = people.team.teamId!!,
                lastName = people.lastName,
                firstName = people.firstName,
                sex = people.sex,
                bloodType = people.bloodType,
                serveYears = people.serveYears,
                role = people.role,
                hometown = people.hometown
            )
        }
    }
}